src/libstd/sys/unix/thread.rs

```diff
@@ -32,6 +32,10 @@ unsafe impl Sync for Thread {}
 impl Thread {
     pub unsafe fn new<'a>(stack: usize, p: Box<FnBox() + 'a>)
                           -> io::Result<Thread> {
+        if cfg!(target_os = "emscripten") {
+            return Err(io::Error::new(io::ErrorKind::Other, "emscripten does not support threads"));
+        }
```

Comment:
Perhaps we could leave this out if it's *only* intended for debugging? In theory once we get panics working better the error messages would be higher quality?